engine/api/src/main/scala/pl/touk/nussknacker/engine/api/typed/typing.scala

```diff
@@ -161,6 +161,10 @@ object typing {
             case (k, v) => k -> fromInstance(v)
           }.toMap
           TypedObjectTypingResult(fieldTypes)
+        case elem :: _ =>
+          TypedClass(obj.getClass, List(fromInstance(elem)))
+        case nonEmptyJavaList: java.util.List[_] if !nonEmptyJavaList.isEmpty =>
+          TypedClass(obj.getClass, List(fromInstance(nonEmptyJavaList.get(0))))
```

Comment:
maybe it would be better to have union of types coming from all elements?

Reply:
I aggree with that

Reply:
Done.